### PR TITLE
[CIAPP-2828] Add GitHub CI Visibility docs

### PR DESCRIPTION
**Pull request by cg14823**
### What does this PR do?

Adds the documentation for tracing GitHub Action workflows with CI Visibility.

### Preview

https://docs-staging.datadoghq.com/carlos.gonzalez/ci-app-gh-docs/continuous_integration/setup_pipelines/github/

### Additional Notes

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

File: content/en/continuous_integration/setup_pipelines/_index.md [hunk 1]
```diff
@@ -6,6 +6,7 @@ disable_sidebar: true
 ---
 
 {{< whatsnext desc="CI pipelines providers:" >}}
+    {{< nextlink href="continuous_integration/setup_pipelines/github" >}}GitHub{{< /nextlink >}}
```
[the remark this answers is not in the document]
Comment (fermayo) [reply content/en/continuous_integration/setup_pipelines/_index.md:11]: Should be OK to leave it as GitHub to be consistent with our in-app onboarding

File: content/en/continuous_integration/setup_pipelines/github.md [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    - link: "/continuous_integration/explore_pipelines"
+      tag: "Documentation"
+      text: "Explore Pipeline Execution Results and Performance"
+    - link: "/continuous_integration/troubleshooting/"
+      tag: "Documentation"
+      text: "Troubleshooting CI"
+---
+
+{{< site-region region="us,eu,us3" >}}
+## Compatibility
```
Comment (fermayo) [inline content/en/continuous_integration/setup_pipelines/github.md:14]:
```suggestion
<div class="alert alert-info">GitHub Actions integration is in beta. There are no billing implications for activating the GitHub Actions integration during this period.
</div>

## Compatibility
```

**Review by fermayo — CHANGES_REQUESTED** [review]

**Comment by fermayo** [comment]
@cg14823 looks like the merge didn't go very well, as there are tons of changes now in this PR that are unrelated

**Comment by cg14823** [comment]
@fermayo  Yes, I just noticed not sure what happened at this point I think it is easier to just copy paste the relevant ones into a new PR.

**Comment by cg14823** [comment]
Closing in favour of https://github.com/DataDog/documentation/pull/12192 as the merge went haywire so it is easier yo just copy paste the relevant stuff than fixing this.

**Pull request closed without merging**